src/harp_pedal_wizard.lua

```diff
@@ -0,0 +1,1506 @@
+            if not found then
+              error = true
+              goto error
+            end
+            scale_deg = scale_deg + 1
+            last = "A"
+          end
+        end -- for i...
+      elseif chd then
+        local ind_string_ltrs = {A_ltr, B_ltr, C_ltr, D_ltr, E_ltr, F_ltr, G_ltr}
+        local ind_string_nums = {A_num, B_num, C_num, D_num, E_num, F_num, G_num}
+        for i, j in pairs(ind_string_ltrs) do
+          if ind_string_ltrs[i][2] ~= root_string then
+            local match = false
+            local count = 0
+            repeat
+              for k, l in pairs(scale_new) do
+                for m,n in pairs(ind_string_nums[i]) do
+                  if ind_string_nums[i][m] == scale_new[k] then
+                    scale_ltrs = scale_ltrs..", "..ind_string_ltrs[i][m]
+                    match = true
+                    goto continue
+                  end
+                end
+              end -- for k, l...
+              count = count + 1
+              if count == 25 then
+                print("Something clearly went wrong...")
+                match = true
+              end
+              ::continue::
+            until ( match == true )
+          end -- if ind_string_ltrs[i][2]...
+        end -- for i, j...
+      end -- if chd...
+
+      if scale == "whole tone" or scale == "major pentatonic" or scale == "minor pentatonic" then
+        chd = false
+      end -- temporary change for exotic scales!
+
+      harp_diagram(scale_ltrs, diag, scaleinfo, partial)
+---
+      ::error::
+      if error then
+        print("That scale won't work.")
+        local str = finale.FCString()
+        str.LuaString = "That scale won't work, sorry. \n Try again using "..enharmonic.LuaString.." "..scale.."?"
+        local result = ui:AlertYesNo(str.LuaString, NULL)
+        if result == 2 then harp_scale(enharmonic.LuaString, scale, diag, chd) end
+      end -- error
+
+    end -- function harp_scale()
+-------
+
+    function harp_dialog()
+--            local path = path_set("com.harp_diagram.txt")
+      local str = finale.FCString()
+      local diag = true
+      local chd = false
+--            local file_r = io.open(path.LuaString, "r")
+
+      if config.chord_check == 0  or config.chord_check == "0" then chd = false 
+      elseif config.chord_check == 1 or config.chord_check == "1" then chd = true end
+      if config.diagram_check == 0  or config.diagram_check == "0" then diag = true 
+      elseif config.diagram_check == 1 or config.diagram_check == "1"  then diag = false end
+
+      local row_y = 0 -- The various controls will use this to consistently place themselves vertically
+--
+      function format_ctrl(ctrl, h, w, st)
+        ctrl:SetHeight(h)
+        ctrl:SetWidth(w)
+        str.LuaString = st
+        ctrl:SetText(str)
+      end -- function format_ctrl
+--
+      local dialog = finale.FCCustomLuaWindow()
+      if nil ~= context.window_pos_x and nil ~= context.window_pos_y then
```

Comment:
A little nit, but `context.window_pos_x ~= nil` and `context.window_pos_y ~= nil` would be better here.

The way you originally wrote it is called a [yoda condition](https://en.wikipedia.org/wiki/Yoda_conditions). While they are valid syntax and there are a few minor advantages of them, they can in general make it harder to read code.

Reply:
I don't remember where I copied this from, but it is definitely from some other script that I stole it from. I never really got this working (retaining the window position), but I'll go ahead and change

Reply:
yoda condition... I just got it!

```
if you_become == 900 then
    look_so_good = false
end
```